compendium/DeclarativeServices/src/metadata/MetadataParserImpl.cpp

```diff
@@ -135,6 +138,81 @@ MetadataParserImplV1::CreateComponentMetadata(const AnyMap& metadata) const
   ObjectValidator(metadata, "name", /*isOptional=*/true)
     .AssignValueTo(compMetadata->name);
 
+  // component.configuration-policy (Optional)
+  compMetadata->configurationPolicy = CONFIG_POLICY_IGNORE;
+  unsigned short configPolicy = 0;
+  object =
+    ObjectValidator(metadata, "configuration-policy", /*isOptional=*/true);
+  if (object.KeyExists()) {
+    object.AssignValueTo(compMetadata->configurationPolicy);
+    configPolicy = 1;
+  }
+
+  // component.configuration-pid (Optional)
+  unsigned short configPid = 0;
+  object = ObjectValidator(metadata, "configuration-pid", /*isOptional=*/true);
+  if (object.KeyExists()) {
+    configPid = 1;
+    if (configPolicy == 1) {
+      const auto configPids =
+        object.GetValue<std::vector<cppmicroservices::Any>>();
+      std::transform(
+        std::begin(configPids),
+        std::end(configPids),
+        std::back_inserter(compMetadata->configurationPids),
+        [](const auto& configPid) {
+          return ObjectValidator(configPid).GetValue<std::string>();
+        });
+
+      //search for a configuration pid equal to $. If present replace with component name.
+      // Also search for duplicates pids. These are errors.
+      std::unordered_map<std::string, std::string> duplicatePids;
+      for (auto& pid : compMetadata->configurationPids) {
+        if (pid == "$") {
+          pid = compMetadata->name;
+        }
+
+        if (duplicatePids.find(pid) != duplicatePids.end()) {
+          std::string msg =
+            "configuration-pid error in the manifest. Duplicate pid detected. ";
+          msg.append(pid);
+          throw std::runtime_error(msg);
+        }
+        duplicatePids.emplace(pid, pid);
+      };
+    }
+  }
+  /* In order to participate in ConfigurationAdmin both the configuration-policy
+     * and the configuration-pid must be present in the manifest.json file. 
+     * Otherwise the configuration-policy is set to ignore. If only one is present
+     * a Warning message is logged.
+     */
+  if (configPolicy ^ configPid) {
+    compMetadata->configurationPolicy = CONFIG_POLICY_IGNORE;
+    compMetadata->configurationPids.clear();
+    std::string msg = "Warning: configuration-policy has been set to ignore.";
+    msg.append(
+      " Both configuration-policy and configuration-pid must be present");
+    msg.append(" to participate in Configuration Admin. ");
+    logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_ERROR, msg);
```

Comment:
Jeff: check to see if this needs to be an error or a warning, and make changes accordingly (comments on line 188 says it should be a warning)